provisioning/baremetal_pod.go

```diff
@@ -612,34 +627,47 @@ func createContainerMetal3Mariadb(images *Images) corev1.Container {
 }
 
 func createContainerMetal3Httpd(images *Images, config *metal3iov1alpha1.ProvisioningSpec, sshKey string) corev1.Container {
-	port, _ := strconv.Atoi(baremetalHttpPort) // #nosec
+	port, _ := strconv.Atoi(baremetalHttpPort)             // #nosec
+	httpsPort, _ := strconv.Atoi(baremetalVmediaHttpsPort) // #nosec
+
+	volumes := []corev1.VolumeMount{
+		sharedVolumeMount,
+		imageVolumeMount,
+		ironicTlsMount,
+		inspectorTlsMount,
+	}
+	if !config.DisableVirtualMediaTLS {
+		volumes = append(volumes, vmediaTlsMount)
+	}
+
 	container := corev1.Container{
 		Name:            "metal3-httpd",
 		Image:           images.Ironic,
 		ImagePullPolicy: "IfNotPresent",
 		SecurityContext: &corev1.SecurityContext{
 			Privileged: pointer.BoolPtr(true),
 		},
-		Command: []string{"/bin/runhttpd"},
-		VolumeMounts: []corev1.VolumeMount{
-			sharedVolumeMount,
-			imageVolumeMount,
-			ironicTlsMount,
-			inspectorTlsMount,
-		},
+		Command:      []string{"/bin/runhttpd"},
+		VolumeMounts: volumes,
 		Env: []corev1.EnvVar{
 			buildEnvVar(httpPort, config),
 			buildEnvVar(provisioningIP, config),
 			buildEnvVar(provisioningInterface, config),
 			buildSSHKeyEnvVar(sshKey),
 			buildEnvVar(provisioningMacAddresses, config),
+			buildEnvVar(vmediaHttpsPort, config),
```

Comment:
Is it OK to set this even when DisableVirtualMediaTLS is set to True?

Reply:
It won't trigger anything in ironic-image (there is a default value for the port anyway). I tried to reduce complexity by having fewer conditionals, but I can change this.